### ISSUE TITLE
GoReleaser Action not working with `ldflag`

### ISSUE THREAD

**Issue by jay-kinder-cts** [goreleaser/goreleaser-action#396]
https://goreleaser.com/cookbooks/using-main.version/

Please let me know if i have misunderstood this, but I was expecting to have a `main.version`  ldflag available to me in my `main.go`.

I have set it up as above, but always receive a `dev` response when using the `-v` flag.

If I have misunderstood, is there a way to get hold of this through the action? Thanks

**Comment by caarlos0** [comment]
you are likely not running it against a tag.

**Comment by jay-kinder-cts** [comment]
This is still causing a problem:

action:

```
name: Release Pipeline
on:
  push:
    tags:
      - "*"

jobs:
  release:
    name: Release New Version
    runs-on: ubuntu-latest

    steps:
      - name: Check Out Code
        uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Set up Go
        uses: actions/setup-go@v3
        with:
          go-version-file: "./go.mod"

      - name: Run GoReleaser
        uses: goreleaser/goreleaser-action@v4
        with:
          distribution: goreleaser
          version: latest
          args: release --clean
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

goreleaser yaml (I was under the impression the ldflag i want was default, but just to be sure i tried this)
```
project_name: grole
builds:
  - env:
      - CGO_ENABLED=0
    goos:
      - linux
      - windows
      - darwin
    goarch:
      - amd64
      - arm64
    ldflags:
      - -s -w -X main.version={{.Version}}
```

main.go

```
var version = "dev"

func main(){
fmt.Printf("version: %s 👍\n", version)
}
```

I've taken lots of functionality out of my main.go, but this is the jist of the issue. This always causes `version: dev 👍` to be printed

Thanks
```

**Comment by caarlos0** [comment]
can you make a minimal reproducible example repository?

**Comment by dwertent** [comment]
@jay-kinder try setting it up like this: `- -s -w -X "main.version={{.Version}}"`

e.g.
```
project_name: grole
builds:
  - env:
      - CGO_ENABLED=0
    goos:
      - linux
      - windows
      - darwin
    goarch:
      - amd64
      - arm64
    ldflags:
      - -s -w -X "main.version={{.Version}}"
```

**Comment by caarlos0** [comment]
@ggpalacio what makes you think its ignored?

fwiw, if it works locally, it should work on the action too, assuming same versions et al.

maybe worth adding a `--verbose` to confirm as well

**Comment by caarlos0** [comment]
I think either you're mistaken or there's something else involved.

Here's a repo copying your configs, and working as expected:

- https://github.com/caarlos0-graveyard/goreleaser-action-issue-396
- https://github.com/caarlos0-graveyard/goreleaser-action-issue-396/actions/runs/13812058555/job/38635713604#step:4:52
- https://github.com/caarlos0-graveyard/goreleaser-action-issue-396/releases/tag/v0.1.0

Downloaded the archive and ran it:

```console
$ wget https://github.com/caarlos0-graveyard/goreleaser-action-issue-396/releases/download/v0.1.0/my-cli_Darwin_arm64.tar.gz
Output hidden for brevity 

$ tar xzvf my-cli_Darwin_arm64.tar.gz
x my-cli

$ ./my-cli
version: 0.1.0 👍
```

**Comment by zisuu** [comment]
Hi, I just came across this when debugging the same problem myself. What I found out is that if I manually download the .tar.gz from my GitHub release, extract it, and then execute that binary with `--version` it works:

```bash
./mybinary --version
Version: 0.3.7
Commit: 4a26fb9
Date: 2025-05-13T13:30:46Z
```

But if I install the package with the command `go install github.com/zisuu/mybinary@v0.3.7` and then run

```bash
mybinary --version
Version: unknown
Commit: unknown
Date: unknown
```

This happens because the `go install` command fetches the source code from the repo and compiles it locally without the `ldflags` unless explicitly provided. @caarlos0, maybe you had the same problem?

I don't know how to fix this, but at least I now understand why it wasn't working as expected, and I can live with that.
